src/cpp/shared/scripting/funcs.cpp

```diff
@@ -15,8 +15,59 @@ extern "C" {
 #include "scripting/deepbinding.h"
 #include "scripting/register.h"
 
+#include "sound/sound.h"
+
+enum SoundEffectWait { SND_DO_WAIT, SND_DONT_WAIT };
+
+static bool PlaySoundEffect(std::string snd, SoundEffectWait wait, SAMPLE2* samp) {
+	SAMPLE2 res = NULL_SAMPLE2;
+	if (!snd.empty()) {
+		char* src;
+		snd += ".82m";
+		src = strdup(snd.c_str());
+		res = LoadPlaySample(src);
+		if (samp != NULL) {
+			memcpy(samp, &res, sizeof(SAMPLE2));
+		}
+		if (wait == SND_DO_WAIT) {
+			WaitEndSample(res, res.sample);
+		}
+		free(src);
+		return true;
+	}
+	return false;
+}
+
+static bool CheckLocationItem(mapCell *loc) {
+	if (loc->objType == LOCATION_ANCIENT_LAMP) {
```

Comment:
Have you tested this? I don't think this works. The high bit is TILE_HAS_EVENT.

Also, a shorter way to test this:

switch (x) {
  case LOCATION_ANCIENT_LAMP:
  case LOCATION_ARTIFACT:
  case LOCATION_RESOURCE
  ...
  case LOCATION_SEA_CHEST:
    return true;
  default:
    return false
}

Reply:
I think that has been addressed. Weird that it was working for me before, but the expression should now correctly use the "objType" field from the map cell.